specs/vep-994-jupyter-notebook-integration/README.md

```diff
@@ -229,9 +237,20 @@ The availability of the extension will be managed by JupyterLab since it is goin
 In terms of security, Jupyter uses tornado to ensure only authorized user can request the Jupyter server
 You can read more [here](https://jupyter-notebook.readthedocs.io/en/stable/security.html#).
 
-VDK Control Service uses authentication in REST API, based on OAuth2 To authenticate specify OAuth2 access token as Authorization/Bearer Header.
-The testing installation uses (Staging) CSP Authentication provider. To get access token you need refresh or access token To get refresh token go to https://console-stg.cloud.vmware.com/csp/gateway/portal/#/user/tokens
-
+VDK Control Service uses authentication in REST API, based on OAuth2 To authenticate specify OAuth2 access token as an Authorization/Bearer Header.
+Access token would be generated using one of 2 approaches depending on how Jupyter server is deployed:
+
+* Standalone JupyterLab (catered for individual users in their local environments):
+   * Login:
+       Initiates the [OAuth2 Authorization Flow](https://tools.ietf.org/html/rfc6749#section-4.1) upon selecting "Login" from the VDK dropdown with callback to the server
+       The server would finish the authorization flow leveraging [tornado Oauth2Mixin](https://www.tornadoweb.org/en/stable/auth.html)
```

Comment:
This definitely requires that the server is publicly visible.

Reply:
No not really . You can see how vdk login (interactive mode) works as in supercollider . It opens a browser and then redirects you to localhost:someport and we listen on that port to get the authorisation code and use it to exchange it for access token. 

It is described in the linked RFC spec but may not be that clear

Reply:
Ah sorry!